### PR TITLE
feat(player): add seek action and handle seeking player state

### PR DESCRIPTION
**Pull request by elpiel**
Core: https://github.com/Stremio/stremio-core/pull/689

For testing:
https://stremio.github.io/stremio-web/feat/seek-player-action/

### CODE REVIEW

**Review by kKaskak — DISMISSED** [review]

**Review by kKaskak — DISMISSED** [review]

File: src/routes/Player/Player.js [hunk 1]
```diff
@@ -158,6 +161,7 @@ const Player = ({ urlParams, queryParams }) => {
     }, []);
 
     const onSeekRequested = React.useCallback((time) => {
+        !seeking && setSeeking(true);
```
Comment (tymmesyde) [inline src/routes/Player/Player.js:164]: this check can be removed `!seeking &&`

File: src/routes/Player/Player.js [hunk 2]
```diff
@@ -553,6 +561,11 @@ const Player = ({ urlParams, queryParams }) => {
                 }
             }
         };
+        const onKeyUp = (event) => {
+            if (event.code === 'ArrowRight' || event.code === 'ArrowLeft') {
+                seeking && setSeeking(false);
```
Comment (tymmesyde) [inline src/routes/Player/Player.js]: this check can be removed `seeking &&`

File: src/routes/Player/Player.js [hunk 3]
```diff
@@ -468,6 +475,7 @@ const Player = ({ urlParams, queryParams }) => {
                     if (!menusOpen && !nextVideoPopupOpen && video.state.paused !== null) {
                         if (video.state.paused) {
                             onPlayRequested();
+                            setSeeking(false);
```
Comment (tymmesyde) [inline src/routes/Player/Player.js:471]: `setSeeking(false);` is already present in `onPlayRequested()`

File: src/routes/Player/Player.js [hunk 4]
```diff
@@ -345,7 +349,10 @@ const Player = ({ urlParams, queryParams }) => {
         if (video.state.time !== null && !isNaN(video.state.time) &&
             video.state.duration !== null && !isNaN(video.state.duration) &&
             video.state.manifest !== null && typeof video.state.manifest.name === 'string') {
-            timeChanged(video.state.time, video.state.duration, video.state.manifest.name);
+            seeking ?
+                seek(video.state.time, video.state.duration, video.state.manifest.name)
+                :
+                timeChanged(video.state.time, video.state.duration, video.state.manifest.name);
         }
     }, [video.state.time, video.state.duration, video.state.manifest]);
```
Comment (tymmesyde) [inline src/routes/Player/Player.js]: `seeking` should be included in the useEffect deps

File: src/routes/Player/Player.js (hunk 1 above)
Comment (tymmesyde) [inline src/routes/Player/Player.js]: this check is unnecessary `!seeking &&`

**Review by tymmesyde — CHANGES_REQUESTED** [review]
If i play the video and start to seek using the seekbar, the seek action will still be sent after stopping

**Review by kKaskak — APPROVED** [review]

**Pull request merged**